### PR TITLE
Replace direct environment variable access with Environment singleton

### LINKED ISSUE #11: API key entry in the app config for etherscan's Api-Key token.
### Create API key entry in the application configuration for etherscan (https://etherscan.io/apis).
Etherscan rate limits to 5 requests a second. Exceeding this figure will result in the app being blocked.

* Include api key config in app generation/tooling.
* API key should be added as an environment variable to the project.
* An API key itself should not be committed to the repo.
  * Instructions on how to generate and load an api key need to be added to the ReadMe/Wiki

### PR DESCRIPTION
**Pull request by dan1elhughes**
### Description

Replaces all `process.env` usage with a single Environment object to control access and be more verbose about missing values

### Issues fixed

Fixes #11 

### Checklist

- [x] `npm test` returns no warnings or errors.

### CODE REVIEW

**Comment by dan1elhughes** [comment]
It can and does, but using a class with a constructor allows us to shout about missing values and provide helpful error messages

**Comment by dan1elhughes** [comment]
It could be a function, but I like the class/constructor paradigm for this

File: src/utils/environment.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/* global process */
+
+class Environment {
+	constructor() {
+		Object.assign(this, ...this.requiredEnvironmentVariables.map(this._loadOrWarn));
```
Comment (ShaunBulbrook) [inline src/utils/environment.js]: Can you leave a comment explaining this line of code.

**Comment by ShaunBulbrook** [comment]
Rebase and we're good to go!

**Pull request merged**